### PR TITLE
Handle track consequence

### PR DESCRIPTION
**Pull request by praveek**
## Description

Handle track consequence 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPAnalytics/Sources/Analytics.swift [hunk 1]
```diff
@@ -56,48 +57,82 @@ public class Analytics: NSObject, Extension {
...
+            Log.warning(label: LOG_TAG, "track - Dropping request (Privacy is opted out).")
             return
         }
 
-        let analyticsVars = processAnalyticsVars(event: event)
-        let analyticsData = processAnalyticsData(event: event)
+        guard let data = data, data.keys.contains(AnalyticsConstants.EventDataKeys.TRACK_STATE) ||
+            data.keys.contains(AnalyticsConstants.EventDataKeys.TRACK_ACTION) ||
+            data.keys.contains(AnalyticsConstants.EventDataKeys.CONTEXT_DATA) else {
+            Log.warning(label: LOG_TAG, "track - Dropping request as event data is missing state, action & contextData")
```
Comment (rymorale) [inline AEPAnalytics/Sources/Analytics.swift]: the log should have an "or" rather than an "&".

File: AEPAnalytics/Sources/AnalyticsHelper.swift [hunk 2]
```diff
@@ -27,6 +32,17 @@ class AnalyticsHelper {
...
     }
+
+    /// Combines the application name, version, and version code into a formatted application identifier
+    /// - Returns: `String` formatted Application identifier
+    static func getApplicationIdentifier() -> String {
+        let applicationName = systemInfoService.getApplicationName() ?? ""
+        let applicationVersion = systemInfoService.getApplicationVersionNumber() ?? ""
+        let applicationBuildNumber = systemInfoService.getApplicationBuildNumber() ?? ""
+        // Make sure that the formatted identifier removes white space if any of the values are empty, and remove the () version wrapper if version is empty as well
+        return "\(applicationName) \(applicationVersion) (\(applicationBuildNumber))".replacingOccurrences(of: "  ", with: " ").replacingOccurrences(of: "()", with: "").trimmingCharacters(in: .whitespacesAndNewlines)
```
Comment (rymorale) [inline AEPAnalytics/Sources/AnalyticsHelper.swift:43]: for this code: 
```swift
.replacingOccurrences(of: "  ", with: " ")
```
do we want the replacement to be `""` instead?

**Review by rymorale — APPROVED** [review]
just 2 small things to take a look at but overall looks good. feel free to merge after taking a look at the comments.

File: AEPAnalytics/Sources/AnalyticsHelper.swift (hunk 2 above)
Comment (praveek) [reply AEPAnalytics/Sources/AnalyticsHelper.swift:43]: If applicationVersion is empty, we still need `" "` in between.

**Pull request merged**